channel/consensus_channel/consensus_channel.go

```diff
@@ -204,22 +192,42 @@ func (a Add) RightDeposit() big.Int {
 
 var ErrIncorrectTurnNum = fmt.Errorf("incorrect turn number")
 
-// Add updates Vars by including a guarantee, updating balances accordingly
-func (vars Vars) Add(p Add) (Vars, error) {
+// Add mutates Vars by
+// - increasing the turn number by 1
+// - including the guarantee
+// - adjusting balances accordingly
+func (vars *Vars) Add(p Add) error {
 	if p.turnNum != vars.TurnNum+1 {
-		return Vars{}, ErrIncorrectTurnNum
+		return ErrIncorrectTurnNum
 	}
 
+	// Increase the turn number
 	vars.TurnNum += 1
 
-	o, err := vars.Outcome.DivertToGuarantee(p)
+	o := vars.Outcome
+	g := p.Guarantee
 
-	if err != nil {
-		return Vars{}, err
+	// Include the guarantee
+	_, found := o.guarantees[g.target]
+	if found {
+		return ErrDuplicateGuarantee
 	}
+	o.guarantees[g.target] = g
```

Comment:
It might be better to delay setting this key until all of the error checking is completed. Otherwise, to recover from an error, the value for the key will need to be removed.

Reply:
b56cf79ed6a8919f46a4205fff1b99d193a250dd